File: Medii de Proiectare si Programare/TrackAndField-C#/TrackAndField/model/Registration.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TrackAndField.model
{
    class Registration : Entity<Tuple<Participant, Track>>
    {
        public Participant Participant { get; set; }
        public Track Track { get; set; }

        public Registration(Participant participant, Track track)
        {
            ID = new Tuple<Participant, Track>(participant, track);
            Participant = participant;
            Track = track;
        }

        public override string ToString()
        {
            return "Participant: " + Participant + " | Track: " + Track;
        }
    }
}

File: Medii de Proiectare si Programare/TrackAndField-Java/src/main/java/model/Registration.java
package model;

public class Registration extends Entity<Tuple<Participant, Track>> {
    private Participant participant;
    private Track track;

    public Registration(Participant participant, Track track) {
        this.setId(new Tuple<>(participant, track));
        this.participant = participant;
        this.track = track;
    }

    public Participant getParticipant() {
        return participant;
    }

    public void setParticipantID(Participant participant) {
        this.participant = participant;
    }

    public Track getTrack() {
        return track;
    }

    public void setTrack(Track track) {
        this.track = track;
    }

    @Override
    public String toString() {
        return "Participant: " + this.participant + " | Track: " + this.track;
    }
}
